cfht2caom2/instruments.py

```diff
@@ -912,125 +947,87 @@ def _update_plane_provenance(self):
             f'End _update_plane_provenance for {self._storage_name.obs_id}'
         )
 
+    def _use_existing_observation(self, bp):
+        """Use values from an existing Observation to set Observation-level blueprint values."""
+        if self._observation is not None:
+            # for the cases where the blueprint keys are not the same as the Python class data member names
+            x = {
+                'ambientTemp': 'ambient_temp',
+                'geoLocationX': 'geo_location_x',
+                'geoLocationY': 'geo_location_y',
+                'geoLocationZ': 'geo_location_z',
+                'metaProducer': 'meta_producer',
+                'metaRelease': 'meta_release',
+                'observationID': 'observation_id',
+                'pi': 'pi_name',
+                'sequenceNumber': 'sequence_number',
+            }
+            for attribute in bp._plan:
+                if attribute.startswith('Observation'):
+                    attribute_names = attribute.split('.')
+                    y = x.get(attribute_names[1], attribute_names[1])
+                    if hasattr(self._observation, y):
+                        attribute_ptr = getattr(self._observation, y)
+                        if len(attribute_names) > 2:
+                            z = x.get(attribute_names[2], attribute_names[2])
+                            if hasattr(attribute_ptr, z):
+                                attribute_ptr = getattr(attribute_ptr, z)
```

Comment:
This is the only difference between `if` and `else`. I would put the next 3 lines outside the `if/else`

Reply:
Moving the next 3 lines outside the `if/else` means the `clear` + `set` calls always occur. These call currently do not occur if the `hasattr` call files.